units/repotracker.go

```diff
@@ -84,11 +86,16 @@ func (j *repotrackerJob) Run() {
 		return
 	}
 
-	ref, err := fetchProjectRefByRepo(j.Owner, j.Repo)
+	ref, err := model.FindOneProjectRefByRepoAndBranch(j.Owner, j.Repo, j.Branch)
 	if err != nil {
 		j.AddError(err)
 		return
 	}
+	if ref == nil {
+		j.AddError(errors.Errorf("Could not find project ref for repo '%s/%s' with branch '%s'",
+			j.Owner, j.Repo, j.Branch))
```

Comment:
presumably we're going to get some number of push events for random branches that people want, and this is just a case to return early without an error.

Reply:
PR #598 has that change already.